app/Member.php

```diff
@@ -4,27 +4,52 @@
 
 use Carbon\Carbon;
 use Collective\Html\Eloquent\FormAccessible;
+use Grimzy\LaravelMysqlSpatial\Eloquent\SpatialTrait;
 use Illuminate\Database\Eloquent\Model;
 use Illuminate\Support\Arr;
+use App\Events\MemberUpdated;
+use App\Jobs\UpdateMemberGeolocation;
 
 class Member extends Model
 {
 
-	use FormAccessible;
+	use FormAccessible, SpatialTrait;
 
 	protected $guarded = ['id', 'created_at', 'updated_at'];
 
 	protected $dates = ['created_at', 'updated_at', 'geboortedatum'];
 
+	protected $spatialFields = ['geolocatie'];
+	
 	// Number of points needed for every level in the points system
 	const RANK_POINTS = [0, 3, 10, 20, 35, 50, 70, 100];
+	
+	protected $dispatchesEvents = [
+		'updated' => MemberUpdated::class  // TODO dispatch event only if address properties have updated?
+	];
 
 	// Full name
 	public function getVolnaamAttribute()
 	{
 		return str_replace('  ', ' ', $this->voornaam . ' ' . $this->tussenvoegsel . ' ' . $this->achternaam);
 	}
 
+	// Full address as single string
+	public function getVolledigAdresAttribute()
+	{
+		return $this->adres . ', ' . $this->postcode . ' ' . $this->plaats;
+	}
+
+	public function getGeolocatieAttribute()
+	{
+		// Call geocoder if location is not yet available
+		if ($this->attributes['geolocatie'] === null) {
+			UpdateMemberGeolocation::dispatchSync($this);
+		}
+
+		return $this->attributes['geolocatie'];  // TODO this still returns null even if geolocatie was updated, how to get the new value?
```

Comment:
Psst. the attributes are cached for the current instance, if you want to fetch the "new" attributes you need to call `refresh()` on the model